docs/src/main/asciidoc/datasource.adoc

```diff
@@ -115,176 +126,113 @@ We currently include these built-in database kinds:
-By using a JDBC driver extension and setting the kind in the configuration,
-Quarkus resolves the JDBC driver automatically,
-so you don't need to configure it yourself.
-If you want to use a database kind that is not part of the built-in ones, use `other` and define the JDBC driver explicitly.
-
+* To use a database kind that is not built-in, use `other` and define the JDBC driver explicitly
++
 [NOTE]
 ====
-You can use any JDBC driver in a Quarkus app in JVM mode (see <<other-databases,Using other databases>>).
-It is unlikely that it will work when compiling your application to a native executable though.
+You can use any JDBC driver in a Quarkus app in JVM mode.
+See <<other-databases,Using other databases>>.
+However, JVM mode is unlikely to work when compiling your application to a native executable.
 
-If you plan to make a native executable, we recommend you use the existing JDBC Quarkus extensions (or contribute one for your driver).
+If you want to build a native executable, use the existing JDBC Quarkus extensions, or contribute one for your driver.
 ====
 
-There is a good chance you will need to define some credentials to access your database.
-
-This is done by configuring the following properties:
-
+. Configure the following properties to define credentials:
++
 [source, properties]
 ----
 quarkus.datasource.username=<your username>
 quarkus.datasource.password=<your password>
 ----
-
++
 You can also retrieve the password from Vault by link:{vault-datasource-guide}[using a credential provider] for your datasource.
 
-Once you have defined the database kind and the credentials, you are ready to configure either a JDBC datasource, a reactive one, or both.
-
-=== JDBC datasource
+Until now, the configuration has been the same regardless of whether you’re using a JDBC or a reactive driver.
+When you have defined the database kind and the credentials, the rest depends on what type of driver you are using.
+It is possible to use JDBC and a reactive driver simultaneously.
 
-JDBC is the most common database connection pattern.
-You typically need a JDBC datasource when using Hibernate ORM.
+==== JDBC datasource
 
-==== Install the Maven dependencies
-
-First, you will need to add the `quarkus-agroal` dependency to your project.
-
-You can add it using a simple Maven command:
+JDBC is the most common database connection pattern, typically needed when used in combination with non-reactive Hibernate ORM.
 
+. Add the `quarkus-agroal` dependency to your project:
++
 [source,bash]
 ----
 ./mvnw quarkus:add-extension -Dextensions="agroal"
 ----
-
++
 [TIP]
 ====
-Agroal comes as a transitive dependency of the Hibernate ORM extension so if you are using Hibernate ORM,
-you don't need to add the Agroal extension dependency explicitly.
+Agroal comes as a transitive dependency of the Hibernate ORM extension.
+Therefore, the Agroal extension dependency does not have to be added explicitly when Hibernate ORM is in use.
```

Comment:
It's also a dependency of any JDBC based extension (the non-reactive ones). So for sake of simplicity, we could omit this whole section.
What is useful for people to know is that Agroal is the implementation of the JDBC connection pool, and they should use it when using a custom JDBC driver (not necessary when using our extensions).

Reply:
Ok, will replace it with this:

Agroal serves as the JDBC connection pool implementation and is designed to be used in conjunction with a custom JDBC driver.

Reply:
> Ok, will replace it with this:
> 
> Agroal serves as the JDBC connection pool implementation and is designed to be used in conjunction with a custom JDBC driver.

No, it's used by the built-in drivers too. The difference is that when you use a built-in JDBC driver extension, you automatically receive the Agroal extension with it. If you use a custom driver, you have to add Agroal explicitly.

Reply:
> Proposal: _Agroal serves as the JDBC connection pool implementation and can be utilized by both custom and built-in JDBC drivers. Using a built-in JDBC driver extension automatically includes the Agroal extension. For custom drivers, Agroal needs to be added explicitly._

+1